### PR TITLE
Define test plan for existing VCH Management APIs

### LINKED ISSUE #7196: Produce test plan for existing vic-machine-service functionality
**Details:**
For 1.3.0, an initial set of functionality was implemented for the `vic-machine-service`. Testing of that functionality was not systematic, and was focused on the functionality that would be used by the UI.

We should step back and product a comprehensive test plan, covering both unit and integration tests, to ensure we have a solid foundation as we refactor and enhance.

**Acceptance Criteria:**

- [ ] Test plan summarizes existing functionality.
- [ ] Test plan explains how to test existing functionality.
- [ ] Test plan identifies gaps where additional tests are needed (including discussion of late-breaking bugs found in 1.3).
- [ ] Issues are open and prioritized for identified gaps.

### PR DESCRIPTION
**Pull request by zjs**
For 1.3.0, an initial set of functionality was implemented for the `vic-machine-service`. Testing of that functionality was not systematic, and was focused on the functionality that would be used by the UI.

Define a comprehensive test plan to ensure coverage and identify gaps.

Detailed test steps are not provided at this stage; these will be added when the tests are implemented.

Resolves #7196

[skip ci]

### CODE REVIEW

File: tests/test-cases/Group23-VIC-Machine-Service/23-02-VCH-List.md [hunk 1]
```diff
@@ -10,17 +10,72 @@ To verify vic-machine-server can return a list of VCHs including the expected in
...
+
+###  1. Create a VCH using the CLI and verify that it is correctly included when listing via the API
+
+
+Concurrency
+-----------
+
+These tests are intended to verify that the API behaves as expected when performing concurrent operations.
+
+###  1. List VCHs while one is being created
```
Comment (AngieCris) [inline tests/test-cases/Group23-VIC-Machine-Service/23-02-VCH-List.md]: Do you think if it's a good idea for us to have separate test files called `concurrency.md` and `concurrency.robot` under `test-cases/Group23-VIC-Machine-Service` for testing concurrent API operations only? So we can put all of the concurrency related testing, including configure and upgrade/rollback into the same test file. 
Similarly is it a good idea to also have `workflow.md` and `workflow.robot` for common user workflows of using API, so we can unify all the workflow based API tests for different methods together?

Besides, implementation wise, do we have an exiting pattern to do concurrent testing with robot? Or is it easier to test it on nightly

File: tests/test-cases/Group23-VIC-Machine-Service/23-02-VCH-List.md [hunk 2]
```diff
@@ -10,17 +10,72 @@ To verify vic-machine-server can return a list of VCHs including the expected in
...
+###  2. Create a VCH and verify it is returned in the list, with correct information
+
+###  3. Create two VCHs and verify both are returned in the list, with correct information for each
+
+(Not yet implemented)
+
+###  4. Create 64 VCHs and verify all are returned in the list, with correct information for each
+
+(Not yet implemented)
+
```
Comment (AngieCris) [inline tests/test-cases/Group23-VIC-Machine-Service/23-02-VCH-List.md]: Do we want to have a test case for listing VCH power state? An example would be, `create three VCHs, power off one of them, list and see if correct power states are returned`

File: tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.md [hunk 3]
```diff
@@ -10,17 +10,108 @@ To verify vic-machine-server can create a VCH with a specified configuration
...
+
+(Not implemented.)
+
+
+Concurrency
+-----------
+
+These tests are intended to verify that the API behaves as expected when performing concurrent operations.
+
+###  1. Attempt to create two VCHs at the same time
```
Comment (AngieCris) [inline tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.md]: Is it a good idea to test this in some separate cases like the following?
1. Attempt to create two VCHs within the same datacenter
2. Attempt to create two VCHs within different datacenter
I'm not sure if the scenario of same/different datacenter is necessary to test here, but just a thought. (also I'm not sure if the VCH name uniqueness is in datacenter level or cluster level)

File: tests/test-cases/Group23-VIC-Machine-Service/23-02-VCH-List.md (hunk 1 above)
Comment (zjs) [reply tests/test-cases/Group23-VIC-Machine-Service/23-02-VCH-List.md]: I think we can reorganize the tests as we implement them. We should find a pattern that meets the expectations of our CI and other developers, while allowing us to have reasonably-sized suites.

**Pull request merged**